source/MRViewer/MRRibbonSchema.h

```diff
@@ -95,7 +95,7 @@ class MRVIEWER_CLASS RibbonSchemaHolder
     };
 
     MRVIEWER_API static std::vector<SearchResult> search( const std::string& searchStr, int* captionCount = nullptr,
-        std::vector<SearchResultWeight>* weights = nullptr );
+        std::vector<SearchResultWeight>* weights = nullptr, std::function<std::string( const std::shared_ptr<RibbonMenuItem>& )> requirementsFunc = {} );
```

Comment:
we should probably introduce parameters structure for this function

Reply:
reworked